src/test/shared/ui/pickerPrompter.test.ts

```diff
@@ -54,6 +58,34 @@ describe('createQuickPick', function () {
         assert.strictEqual(prompter.quickPick.enabled, true)
         assert.deepStrictEqual(prompter.quickPick.items, items)
     })
+
+    it('creates a new prompter when given an AsyncIterable', async function () {
+        let r1!: (v?: any) => void
+        let r2!: (v?: any) => void
+        const p1 = new Promise(r => (r1 = r))
+        const p2 = new Promise(r => (r2 = r))
+
+        async function* generator() {
```

Comment:
Nit: could you just mock a generator with sinon? Something like `onFirstCall({ data: items[0], done: false }).onSecondCall({ data: items[1], done: true })`

Reply:
I think you can, though I'm in favor of using real generator functions here. With the stubs, it's not clear that we're faking a generator unless you know how the generators work internally